### PR TITLE
Feature/setup

### PR DESCRIPTION
**Pull request by ame-test**
Platform hub docker compose setup.
* Uses `host.docker.internal` for localhost.
  * `localhost` is not possible due to issue with Mac computes with `network_mode: host` in docker compose.
  * - https://stackoverflow.com/questions/55851632/docker-compose-network-mode-host-not-working

README.md documentation added.
- can docker compsoe when environment variables for `.env.local` for both api and auth proxy are done.

### CODE REVIEW

File: platform-hub-web/web.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+FROM node:6.9.1
+
+WORKDIR /app
+COPY . .
+
+RUN npm install -g yarn@0.27.5
+RUN git config --global url."https://".insteadOf git://
+RUN yarn install
```
Comment (jremes-ho) [inline platform-hub-web/web.Dockerfile:8]: This may actually require `gulp` CLI tool.

File: platform-hub-api/api.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+RUN apk update
+RUN apk upgrade
+RUN apk --update add ca-certificates libressl
+RUN update-ca-certificates
+RUN apk --update add bash curl libstdc++ tzdata postgresql-client postgresql-dev
+RUN apk --update add --virtual build_deps sudo build-base libc-dev libressl-dev zlib-dev
+RUN echo 'gem: --no-document' > /etc/gemrc
+
+RUN gem install bundler -v 2.3.27
```
Comment (jremes-ho) [inline platform-hub-api/api.Dockerfile:14]: This may not work with `rbenv`, but may work with default setup.

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1,2 @@
+esdata
+pgdata
```
Comment (jremes-ho) [inline .gitignore:2]: Please add newline after this.

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    volumes:
+      - ./platform-hub-api/.env.local:/platform-hub-api/.env.local
+    depends_on:
+      - platform-hub-db
+    ports:
+      - 8081:8081
+    
+volumes:
+  pgdata:
+  esdata:
```
Comment (jremes-ho) [inline docker-compose.yml:74]: Please add newline.

File: platform-hub-api/api.Dockerfile [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+RUN bundle
+
+CMD [ "bin/rails", "server" ]
```
Comment (jremes-ho) [inline platform-hub-api/api.Dockerfile:17]: Newline here please.

File: platform-hub-web/web.Dockerfile [hunk 6]
```diff
@@ -0,0 +1,10 @@
...
+
+CMD [ "yarn", "run", "serve" ]
```
Comment (jremes-ho) [inline platform-hub-web/web.Dockerfile:10]: Newline here please.

**Review by jremes-ho — COMMENTED** [review]
Few comments on initial review. Thanks.

File: platform-hub-web/web.Dockerfile (hunk 1 above)
Comment (ame-test) [reply platform-hub-web/web.Dockerfile:8]: The `gulp` not found issue has been resolved.
* Was due to `docker-compose.yml` initial volume not allow the `node_modules` folder to exist. See [stackoverflow post](https://stackoverflow.com/questions/30043872/docker-compose-node-modules-not-present-in-a-volume-after-npm-install-succeeds#:~:text=This%20happens%20because%20you%20have,the%20dependencies%20are%20installed%20there.).

File: platform-hub-api/api.Dockerfile (hunk 2 above)
Comment (ame-test) [reply platform-hub-api/api.Dockerfile:14]: I don't understand the comment e.g. what is the issue with the code not using `rbenv`?

File: .gitignore (hunk 3 above)
Comment (ame-test) [reply .gitignore]: Removed file, since it is unnecessary.

File: docker-compose.yml (hunk 4 above)
Comment (ame-test) [reply docker-compose.yml]: added in new line.
* Is there is a specific reason to add the new line at the end of the file or is it just a convention?

File: platform-hub-api/api.Dockerfile (hunk 5 above)
Comment (ame-test) [reply platform-hub-api/api.Dockerfile]: added in new line.

File: platform-hub-web/web.Dockerfile (hunk 6 above)
Comment (ame-test) [reply platform-hub-web/web.Dockerfile]: added in new line.

**Review by jremes-ho — APPROVED** [review]
LGTM ✅

**Review by tom-haynes — APPROVED** [review]
Tested with Rancher desktop on M1, looks good to me:
- Local environment comes up, crud operations possible with local postgres, auth proxy works with old keycloak https://sso-dev.notprod.homeoffice.gov.uk
- bin/rails console does not work in the API container but as discussed this will be looked into at a later date.

**Pull request merged**